### ISSUE TITLE
Use of GetAwaiter().GetResult() causes deadlock inside specflow tests using SingleNodeReplicaSet

### ISSUE THREAD

**Issue by rbev** [wassim-k/MongoSandbox#2]
While updating my test code to use MongoSandbox i've come across a deadlock introduced by some changes to `MongodProcess.cs`

I believe it's a due to the use of `GetAwaiter().GetResult()` in `ConfigureAndWaitForReplicaSetReadiness()`

Here is the minimal replication
https://github.com/rbev/MongoSandboxIssueReplication/tree/4426f20576c089e0cd03c53a048285a5a645cf3a

```csharp
    [When(@"specflow starts a MongoSandbox")]
    public async Task WhenSpecflowStartsAMongoSandbox()
    {
        await Task.Delay(200);
        using var db = MongoRunner.Run(new MongoRunnerOptions()
        {
            UseSingleNodeReplicaSet = true
        });
    }
```

**Comment by rbev** [comment]
I'll note that a workaround seems to be to wrap the call in another task, so it might just need `ConfigureAwait(false)` added to it.

**Comment by wassim-k** [comment]
Thanks @rbev for raising the issue, it has been fixed in v1.0.1
